packages/flutter_availability/lib/src/ui/widgets/input_fields.dart

```diff
@@ -0,0 +1,58 @@
+import "package:flutter/material.dart";
+import "package:flutter_availability/src/util/scope.dart";
+
+/// An input field for time selection
+class TimeInputField extends StatelessWidget {
+  ///
+  const TimeInputField({
+    required this.initialValue,
+    required this.onTimeChanged,
+    super.key,
+  });
+
+  ///
+  final DateTime? initialValue;
+
+  ///
+  final void Function(DateTime) onTimeChanged;
+
+  @override
+  Widget build(BuildContext context) {
+    var theme = Theme.of(context);
+    var availabilityScope = AvailabilityScope.of(context);
+    var options = availabilityScope.options;
+    var translations = options.translations;
+    return TextFormField(
+      decoration: InputDecoration(
+        suffixIcon: const Icon(Icons.access_time),
+        hintText: translations.time,
+        hintStyle: theme.inputDecorationTheme.hintStyle,
+        border: OutlineInputBorder(
+          borderRadius: BorderRadius.circular(10),
+        ),
+      ),
+      initialValue: initialValue != null
+          ? translations.timeFormatter(context, initialValue!)
+          : null,
+      readOnly: true,
+      style: options.textStyles.inputFieldTextStyle,
+      onTap: () async {
```

Comment:
I know it's just a single widget, but let's still keep the functions out of the tree.